pkg/cri/server/sandbox_run.go

```diff
@@ -222,12 +185,32 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 	if err != nil {
 		return nil, fmt.Errorf("failed to create containerd container: %w", err)
 	}
+
+	// Add sandbox into sandbox store in INIT state.
+	sandbox.Container = container
+
 	defer func() {
-		if retErr != nil {
+		if retErr != nil && cleanupErr != nil {
+			log.G(ctx).WithError(cleanupErr).Errorf("there was error during resource cleanup and the sandbox will be put into store")
+
+			sandbox.Status = sandboxstore.StoreStatus(
+				sandboxstore.Status{
+					State: sandboxstore.StateUnknown,
+				},
+			)
+			if err := c.sandboxStore.Add(sandbox); err != nil {
```

Comment:
I think we don't need to patch-update the sandbox's status here, because  
the original status is `StateUnknown` and only updated before adding into sandboxStore. 

I only concern that it seems that there is no validation about `ProcessLabel` before commit(into sandboxStore).
If it happens, we can't persist the record in memory. I think we should pre-check the `ProcessLabel` is valid at the beginning. cc @mikebrow 

REF: https://github.com/containerd/containerd/blob/v1.6.0/pkg/cri/store/sandbox/sandbox.go#L88 about commit

Reply:
The sandboxStore will call the selinux [label.NewContext][1] before caching the sandbox record. 
If the `label.NewContext` returns error, we will lost the record when failed to rollback,

After scan the code again, the selinux label has validation in [here][2]. So this case won't happen,

[1]: <https://github.com/containerd/containerd/blob/v1.6.0/pkg/cri/store/sandbox/sandbox.go#L88>
[2]: <https://github.com/containerd/containerd/blob/v1.6.0/pkg/cri/server/sandbox_run_linux.go#L120>